tchannel/testing/vcr/cassette.py

```diff
@@ -0,0 +1,101 @@
+from __future__ import absolute_import
+
+from itertools import chain
+from collections import deque
+
+from . import types
+from . import yaml
+from .exceptions import RequestNotFoundError, UnsupportedVersionError
+
+
+# Version of the storage format.
+VERSION = 1
+
+
+class Cassette(object):
+    """Represents a series of recorded interactions."""
+
+    def __init__(self, path):
+        """Initialize a new cassette.
+
+        :param path:
+            File path at which this cassette will be stored.
+        """
+        self.path = path
+
+        # Number of times interactions from this cassette have been played.
+        self.play_count = 0
+
+        self._available = deque()
+        self._played = deque()
+        self._recorded = deque()
+
+        self._load()
+
+    @property
+    def data(self):
+        """Get all known data for this cassette."""
+
+        return deque(
+            chain(self._played, self._available, self._recorded)
+        )
+
+    def _load(self):
+        try:
+            with open(self.path, 'r') as f:
```

Comment:
If you're in the process of still writing your test, you shouldn't be using VCR. Once you've written your test, you add VCR on it so that you're no longer making actual requests to the service. If at some point you change the behavior in the test so that the recording is no longer valid, it's on you to modify/delete the recording. A `make clean` command that deletes your test data doesn't make much sense IMO.

Also, some of your concerns around workflow may be alleviated by the Record Modes feature. See #60. Also have a look at vcrpy's documentation when you get a chance.